### PR TITLE
Update trunk to version 4.2

### PR DESCRIPTION
**Pull request by michaelsembwever**

### CODE REVIEW

File: upgrade_tests/upgrade_manifest.py [hunk 1]
```diff
@@ -27,7 +27,8 @@
 CASSANDRA_3_11 = '3.11'
 CASSANDRA_4_0 = '4.0'
 CASSANDRA_4_1 = '4.1'
-TRUNK = CASSANDRA_4_1
+CASSANDRA_4_2 = '4.1'
```
Comment (ekaterinadimitrova2) [inline upgrade_tests/upgrade_manifest.py]: CASSANDRA_4_2 = '4.2' ?

File: upgrade_tests/upgrade_manifest.py [hunk 2]
```diff
@@ -154,14 +157,17 @@ def clone_with_local_env_version(self):
...
-current_3_0_x = VersionMeta(name='current_3_0_x', family=CASSANDRA_3_0, variant='current', version='3.0.24', min_proto_v=3, max_proto_v=4, java_versions=(8,))
+current_3_0_x = VersionMeta(name='current_3_0_x', family=CASSANDRA_3_0, variant='current', version='3.0.26', min_proto_v=3, max_proto_v=4, java_versions=(8,))
 
 indev_3_11_x = VersionMeta(name='indev_3_11_x', family=CASSANDRA_3_11, variant='indev', version='github:apache/cassandra-3.11', min_proto_v=3, max_proto_v=4, java_versions=(8,))
-current_3_11_x = VersionMeta(name='current_3_11_x', family=CASSANDRA_3_11, variant='current', version='3.11.10', min_proto_v=3, max_proto_v=4, java_versions=(8,))
+current_3_11_x = VersionMeta(name='current_3_11_x', family=CASSANDRA_3_11, variant='current', version='3.11.12', min_proto_v=3, max_proto_v=4, java_versions=(8,))
 
 indev_4_0_x = VersionMeta(name='indev_4_0_x', family=CASSANDRA_4_0, variant='indev', version='github:apache/cassandra-4.0', min_proto_v=3, max_proto_v=4, java_versions=(8,))
 current_4_0_x = VersionMeta(name='current_4_0_x', family=CASSANDRA_4_0, variant='current', version='4.0.3', min_proto_v=4, max_proto_v=5, java_versions=(8,))
```
Comment (ekaterinadimitrova2) [inline upgrade_tests/upgrade_manifest.py:166]: It's a pity this will have to be updated in three days...

File: upgrade_tests/upgrade_manifest.py [hunk 3]
```diff
@@ -154,14 +157,17 @@ def clone_with_local_env_version(self):
...
 
+indev_4_1_x = VersionMeta(name='indev_4_1_x', family=CASSANDRA_4_1, variant='indev', version='github:apache/cassandra-4.1', min_proto_v=3, max_proto_v=4, java_versions=(8,))
+#current_4_1_x = VersionMeta(name='current_4_1_x', family=CASSANDRA_4_1, variant='current', version='4.1-alpha1', min_proto_v=4, max_proto_v=5, java_versions=(8,))
```
Comment (ekaterinadimitrova2) [inline upgrade_tests/upgrade_manifest.py:169]: Intentionally commented out?

Comment (michaelsembwever) [reply upgrade_tests/upgrade_manifest.py:169]: yes, not release yet. comment is a placeholder.

**Comment by michaelsembwever** [comment]
merged manually with https://github.com/apache/cassandra-dtest/commit/3801a9b00884279a3e8c095b1cf6fb920ac76fa8

**Pull request closed without merging**